arangod/Aql/Executor/EnumerateListExecutor.cpp

```diff
@@ -82,12 +83,20 @@ class EnumerateListExpressionContext final : public QueryExpressionContext {
                bool& mustDestroy) -> AqlValue {
           mustDestroy = doCopy;
           auto const searchId = variable->id;
-          if (searchId == _outputVariableId) {
+          if (searchId == _outputVariables[0]->id) {
             if (doCopy) {
-              return _currentValue.clone();
+              return _currentValueK.clone();
             }
-            return _currentValue;
+            return _currentValueK;
           }
+          if (_outputVariables.size() > 1 &&
+              searchId == _outputVariables[1]->id) {
+            if (doCopy) {
+              return _currentValueV.clone();
+            }
+            return _currentValueV;
+          }
+
```

Comment:
This code can be generalized in a better way. We already allow a vector of variables, we might as well allow a vector of current values and return for the ith variable the ith value.